### PR TITLE
BUGFIX: SD-908 keep zoom after clicking items on category legend

### PR DESCRIPTION
**Pull request by dlcd2203**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2975
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2688

# Related Jira tasks

SD-908: https://jira.intgdc.com/browse/SD-908

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6843/ : SUCCESS in 3m 47s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2713/ : SUCCESS in 4s

File: src/components/core/geoChart/GeoChartRenderer.tsx [hunk 1]
```diff
@@ -75,11 +75,12 @@ export default class GeoChartRenderer extends React.Component<IGeoChartRendererP
             return;
         }
 
-        this.updateMapWithConfig(prevConfig);
-
         if (selectedSegmentItems && !isEqual(selectedSegmentItems, prevSelectedSegmentItems)) {
             this.setFilterMap();
+            return;
```
Comment (ptquang86) [inline src/components/core/geoChart/GeoChartRenderer.tsx:80]: we can combine this :)
`return this.setFilterMap(); `

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3574/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2714/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2945/ : SUCCESS in 15m 25s

File: src/components/core/geoChart/GeoChartRenderer.tsx (hunk 1 above)
Comment (dlcd2203) [reply src/components/core/geoChart/GeoChartRenderer.tsx]: fixed

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1512.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6845/ : FAILURE in 13m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2716/ : SUCCESS in 3s

**Comment by dlcd2203** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6846/ : SUCCESS in 2m 53s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2718/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2717/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2946/ : SUCCESS in 19m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6848/ : SUCCESS in 12m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2721/ : SUCCESS in 4s

**Comment by naibinh** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3580/ : SUCCESS in 15m 06s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6858/ : SUCCESS in 3m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1667/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2409/ : SUCCESS in 26m 18s

**Pull request merged**